sh_scrapy/settings.py

```diff
@@ -1,35 +1,52 @@
+import logging
 import sys, os, tempfile
 from sh_scrapy.compat import to_native_str, is_string
+from scrapy.settings import BaseSettings
+from scrapy.utils.misc import load_object
 from scrapy.utils.project import get_project_settings
 
 
+logger = logging.getLogger(__name__)
 REPLACE_ADDONS_PATHS = {
     "hworker.bot.ext.page.PageStorageMiddleware":
         "scrapy_pagestorage.PageStorageMiddleware",
     "hworker.bot.ext.persistence.DotScrapyPersistence":
         "scrapy_dotpersistence.DotScrapyPersistence",
 }
+SLYBOT_SPIDER_MANAGER = 'slybot.spidermanager.ZipfileSlybotSpiderManager'
 
 try:
     from scrapy.utils.deprecate import update_classpath
 except ImportError:
     update_classpath = lambda x: x
 
 
-def _update_settings(o, d):
-    # We need to convert settings to string since the S3 download handler
-    # doesn't work if the AWS keys are passed as unicode. Other code may also
-    # depend on settings being str. TODO: we should test this
+def _update_settings(o, d, priority='default'):
+    """
+    We need to convert settings to string since the S3 download handler
+    doesn't work if the AWS keys are passed as unicode. Other code may also
+    depend on settings being str.
+
+    :param o: auxiliary BaseSettings object to merge provided settings
+    :type o: scrapy.settings.BaseSettings instance
+
+    :param d: final Settings object to run a job
+    :type d: scrapy.settings.Settings instance
+    """
     for k, v in d.items():
         d[to_native_str(k)] = to_native_str(v) if is_string(v) else v
-    o.update(d)
+    o.update(d, priority=priority)
 
 
 def _load_autoscraping_settings(s, o):
-    o.setdefault('SPIDER_MANAGER_CLASS', 'slybot.spidermanager.ZipfileSlybotSpiderManager')
-    o.setdefault('SLYCLOSE_SPIDER_ENABLED', True)
-    o.setdefault('ITEM_PIPELINES', {})['slybot.dupefilter.DupeFilterPipeline'] = 0
-    o.setdefault('SLYDUPEFILTER_ENABLED', True)
+    settings = {'ITEM_PIPELINES': {},
+                'SLYDUPEFILTER_ENABLED': True,
+                'SLYCLOSE_SPIDER_ENABLED': True,
+                'SPIDER_MANAGER_CLASS': SLYBOT_SPIDER_MANAGER}
+    for key, value in settings.items():
+        if key not in o:
+            o.set(key, value)
```

Comment:
Right, I think the whole `_load_autoscraping_settings` could be replaced with applying autoscraping settings in `_maybe_load_autoscraping_project` with 0 priority.